features_and_tools/cds-hooks.rst

```diff
@@ -0,0 +1,342 @@
+                            { "patientToGreet", "Patient/{{context.patientId}}" }
+                        },
+                        Hook = "patient-view",
+                        UsageRequirements = "none"
+                    }
+                ));
+            } 
+        }
+
+.. container:: toggle
+
+    .. container:: header
+
+      PatientViewTestHookConfiguration.cs
+      
+    Configuration works the same way as for any other Vonk plugin. You register both the service itself and the contributor that adds the service to the CDS Hooks Discovery document.
+
+    .. code-block:: csharp
+    
+        using System.Diagnostics.CodeAnalysis;
+        using Microsoft.AspNetCore.Builder;
+        using Microsoft.Extensions.DependencyInjection;
+        using Microsoft.Extensions.DependencyInjection.Extensions;
+        using Vonk.Core.Common;
+        using Vonk.Core.Pluggability;
+        
+        namespace Vonk.Plugin.CdsHooks.PatientViewTestHook;
+        
+        [VonkConfiguration(order: 5500, isLicensedAs: VonkConstants.Plugins.Fhir.Operation.CdsHooks)]
+        [Experimental("CdsHooks")]
+        public static class PatientViewTestHookConfiguration
+        {
+            public static IServiceCollection ConfigureServices(this IServiceCollection services)
+            {
+                services.TryAddSingleton<ICdsHooksDiscoveryDocumentContributor, PatientViewTestHookContributor>();
+                services.TryAddScoped<PatientViewTestHookService>();
+                return services;
+            }
+        
+            public static IApplicationBuilder Configure(IApplicationBuilder builder)
+            {
+                builder.OnCdsHooksRequest("patient-view-test-hook")
+                    .HandleAsyncWith<PatientViewTestHookService>((svc, ctx) => svc.HandlePatientViewHook(ctx));
+                return builder;
+            }
+        }
+
+CDS Hooks in FHIR
+-----------------
+
+CDS Hooks structures like the Discovery document, the request and response are not defined in terms of FHIR.
+However, to fit them into the FHIR ecosystem, Firely Server uses the FHIR R4 resource types ``CDSHooksRequest`` and ``CDSHooksResponse`` to represent the response of a CDS Hooks service. 
+These resource types are delivered with Firely Server through the ``errata.zip`` for FHIR R4, and hence also in the pre-built ``vonkadmin.db`` database.
+Note however that:
+
+* Neither of these StructureDefinitions are part of the FHIR specification. They are only available experimentally as logical models in the `FHIR tools package <https://simplifier.net/packages/hl7.fhir.uv.tools.r4>`_.
+* Since logical models do not define resource types, Firely has adjusted those to the StructureDefinitions that are packaged with the server.
+* ``CDSHooksRequest`` has specific elements for each hook, like ``patientToGreet`` for the example service. If a new hook requires additional elements, these should be added to the ``CDSHooksRequest`` resource type.
```

Comment:
Should we elaborate on this further and point out that prefetch elements are always key value pairs and that services need to pick their name and then update the SD? Same for context elements.